stdlib/Distributed/src/remotecall.jl

```diff
@@ -91,8 +96,9 @@ function finalize_ref(r::AbstractRemoteRef)
                     send_del_client_no_lock(r)
                 else
                     # send_del_client only if the reference has not been set
-                    r.v === nothing && send_del_client_no_lock(r)
-                    r.v = nothing
+                    v_cache = @atomic :acquire r.v
+                    v_cache === nothing && send_del_client_no_lock(r)
+                    @atomic :release r.v = nothing
```

Comment:
no other thread has access, since this is inside the finalizer
```suggestion
                    v_cache = @atomic :monotonic r.v
                    v_cache === nothing && send_del_client_no_lock(r)
                    @atomic :monotonic r.v = nothing
```